src/atoms/s2SearchState.ts

```diff
@@ -4,18 +4,33 @@ import { S2SearchResult } from '../api/api-types';
 import { removeProjectReferences } from '../api/referencesAPI';
 import { getS2ReferencesByKeyword, postS2Reference } from '../api/s2SearchAPI';
 import { refreshFileTreeAtom } from './fileExplorerActions';
-import { loadReferencesAtom } from './referencesState';
+import { getReferencesAtom, loadReferencesAtom } from './referencesState';
 
 // #####################################################################################
 // Internal Atoms
 // #####################################################################################
 export const searchResultsAtom = atom<S2SearchResult[]>([]);
+export const paperIdLookupAtom = atom<Record<string, string>>({});
 
 // #####################################################################################
 // References API
 // #####################################################################################
 
 export const getSearchResultsAtom = atom((get) => Object.values(get(searchResultsAtom)));
+export const getPaperIdLookupAtom = atom((get) => get(paperIdLookupAtom));
+
+export const generatePaperIdLookupAtom = atom(null, (get, set) => {
```

Comment:
@jvoytek I don't think we want to do this. For now, the s2 modal should only be able to match the references uploaded via the modal.

Also, the sourceId for existing references aren't their ID.